src/Psalm/Internal/PhpVisitor/Reflector/ClassLikeDocblockParser.php

```diff
@@ -255,21 +266,24 @@ public static function parse(
         if (isset($parsed_docblock->tags['psalm-suppress'])) {
             foreach ($parsed_docblock->tags['psalm-suppress'] as $offset => $suppress_entry) {
                 foreach (DocComment::parseSuppressList($suppress_entry) as $issue_offset => $suppressed_issue) {
-                    $info->suppressed_issues[$issue_offset + $offset + $comment->getStartFilePos()] = $suppressed_issue;
+                    $info->suppressed_issues[$issue_offset + $offset] = $suppressed_issue;
                 }
             }
         }
 
-        $imported_types = array_merge(
-            $parsed_docblock->tags['phpstan-import-type'] ?? [],
-            $parsed_docblock->tags['psalm-import-type'] ?? []
-        );
+        $imported_types = ($parsed_docblock->tags['phpstan-import-type'] ?? []) +
+            ($parsed_docblock->tags['psalm-import-type'] ?? []);
```

Comment:
Good catch, I believe `array_merge()` was destroying offsets here.

Reply:
yeah it was, that was a pain to pinpoint :)